client/components/marketing-survey/cancel-purchase-form/step-components/upsell-step.tsx

```diff
@@ -1,155 +1,277 @@
 import { Button } from '@wordpress/components';
-import { useTranslate } from 'i18n-calypso';
-import rocketImage from 'calypso/assets/images/customer-home/illustration--rocket.svg';
-import pluginsThemesImage from 'calypso/assets/images/customer-home/illustration--task-connect-social-accounts.svg';
-import downgradeImage from 'calypso/assets/images/customer-home/illustration--task-earn.svg';
+import { useTranslate, numberFormat } from 'i18n-calypso';
+import page from 'page';
+import { useState } from 'react';
+import { useSelector } from 'react-redux';
+import imgBuiltBy from 'calypso/assets/images/cancellation/built-by.png';
+import imgBusinessPlan from 'calypso/assets/images/cancellation/business-plan.png';
+import imgFreeMonth from 'calypso/assets/images/cancellation/free-month.png';
+import imgLiveChat from 'calypso/assets/images/cancellation/live-chat.png';
+import imgMonthlyPayments from 'calypso/assets/images/cancellation/monthly-payments.png';
+import imgSwitchPlan from 'calypso/assets/images/cancellation/switch-plan.png';
 import FormattedHeader from 'calypso/components/formatted-header';
-import BusinessATStep from './business-at-step';
-import DowngradeStep from './downgrade-step';
-import FreeMonthOfferStep from './free-month-offer-step';
-import UpgradeATStep from './upgrade-at-step';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import useHappyChat from '../use-happychat';
+import type { UpsellType } from '../get-upsell-type';
 import type { SiteDetails } from '@automattic/data-stores';
 import type { Purchase } from 'calypso/lib/purchases/types';
+import type { TranslateResult } from 'i18n-calypso';
 
 type UpsellProps = {
 	children: React.ReactChild;
 	image: string;
+	title: TranslateResult;
+	acceptButtonText: TranslateResult;
+	acceptButtonUrl?: string;
+	declineButtonText?: TranslateResult;
+	onAccept?: () => void;
+	onDecline: () => void;
 };
 
-function Upsell( { children, image }: UpsellProps ) {
+function Upsell( { image, ...props }: UpsellProps ) {
 	const translate = useTranslate();
+	const declineButtonText = props.declineButtonText || translate( 'Cancel my current plan' );
+	const [ busyButton, setBusyButton ] = useState( '' );
 
 	return (
-		<div className="cancel-purchase-form__upsell-container">
-			<FormattedHeader brandFont headerText={ translate( 'Here is an idea' ) } />
-			<div className="cancel-purchase-form__upsell">
+		<div className="cancel-purchase-form__upsell">
+			<div className="cancel-purchase-form__upsell-content">
+				<div className="cancel-purchase-form__upsell-subheader">
+					{ translate( 'Here is an idea' ) }
+				</div>
+				<FormattedHeader brandFont headerText={ props.title } />
+				<div className="cancel-purchase-form__upsell-text">{ props.children }</div>
+				<div className="cancel-purchase-form__upsell-buttons">
+					<Button
+						isPrimary
+						href={ props.acceptButtonUrl }
+						onClick={ () => {
+							setBusyButton( 'accept' );
```

Comment:
Hey! 👋 I found an issue with this component while working on #78711. Essentially, if `href` is provided, then `a` is rendered instead of a button. This means `disabled` cannot be accepted. However, I see we have an `onClick` method as well.

There are two conflicting solutions. Could you let me know which should be used?

1. Remove the `href` because it isn't a link
2. Remove `disabled` because it isn't a button